internal/grpc/services/publicstorageprovider/publicstorageprovider.go

```diff
@@ -953,45 +923,70 @@ func (s *service) GetQuota(ctx context.Context, req *provider.GetQuotaRequest) (
 	return nil, gstatus.Errorf(codes.Unimplemented, "method not implemented")
 }
 
-// resolveToken returns the path and share for the publicly shared resource.
-func (s *service) resolveToken(ctx context.Context, token string) (*link.PublicShare, *provider.ResourceInfo, *rpc.Status, error) {
-	driver, err := pool.GetGatewayServiceClient(s.conf.GatewayAddr)
+// resolveToken returns the resource info for the publicly shared resource.
+func (s *service) resolveToken(ctx context.Context, share interface{}) (*provider.ResourceInfo, interface{}, error) {
```

Comment:
I'm afraid it needs to be the generic interface. It's being passed around down to `func addShare(i *provider.ResourceInfo, share interface{}) error` where both public and ocm shares get json-marshalled into the opaque map.
I currently can't think of another way except for returning the json directly which I would like to avoid because marshalling into some (arbitrary) format isn't the scope of this method and it's also not clear whether the marshalling actually always has to happen.
Please let me know if you got an idea how to improve.

Reply:
:+1: Thanks for explanation! Fine for me so far...